segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,229 @@
+            new_frames[i] = downsampled_bin[0]
+        elif strategy == 'last':
+            new_frames[i] = downsampled_bin[len(downsampled_bin) - 1]
+    return new_frames
+
+
+def get_transformed_center(coordinate_pair: Tuple[int, int],
+                           box_size: Tuple[int, int],
+                           video_shape: Tuple[int, int]) -> Tuple[int, int]:
+    """
+    Returns a coordinate pair that has been transformed to fit as closely to
+    original position as possible but still with box contained totally in video.
+    This is made to shift the coordinate if the subset box goes out of bound of
+    the video shape. The shift moves the box fully into the frame while maintaining
+    the center of the visual box is as close as possible to the original center.
+    (example coordinate pair at 0, with box size 3 x 3)
+    x x x x    x x x x
+    x x x x    x x x x
+    x x x x -> x x x x
+    x x x x    x 0 x x
+    x 0 x x    x x x x
+    Args:
+        coordinate_pair: the original coordinate location for centering
+        box_size: the size of the box for subset
+        video_shape: the x and y dimensions of the video
+
+    Returns:
+        transformed_center: returns the best fitting center of the box for out
+        of bounds coordinates
+
+    """
+    if box_size[0] > video_shape[0] or box_size[1] > video_shape[1]:
+        raise ValueError('Box is larger or equal in size in one or more'
+                         'dimensions to original video shape. Provide a '
+                         'box size smaller than original video.')
+    else:
+        # if the box size is less than frame size it can only be out of bounds in
+        # direction left and right and one direction up and down
+        transformed_x = coordinate_pair[0]
+        transformed_y = coordinate_pair[1]
+        # can't be centered left
+        if coordinate_pair[0] < (math.ceil(box_size[0]/2) - 1):
+            transformed_x = math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered up
+        if coordinate_pair[1] < (math.ceil(box_size[1]/2) - 1):
+            transformed_y = math.ceil(box_size[1] / 2) - 1
+
+        # can't be centered right
+        if (coordinate_pair[0] + (math.ceil(box_size[0]/2))) > video_shape[0] - 1:
+            transformed_x = video_shape[0] - math.floor(box_size[0] / 2) - 1
+
+        # can't be centered down
+        if (coordinate_pair[1] + (math.ceil(box_size[1]/2))) > video_shape[1] - 1:
+            transformed_y = video_shape[1] - math.floor(box_size[1] / 2) - 1
+
+    return transformed_x, transformed_y
+
+
+def get_centered_coordinate_box_video(coordinate_pair: Tuple[int, int],
+                                      box_size: Tuple[int, int],
+                                      video_array: np.ndarray):
+    """
+    Function to get video subset centered around coordinate pair with
+    size specified by box size tuple. Function takes a subset of each frame
+    and stacks together to get final video.
```

Comment:
I think your docstring needs to specify row/col or x/y for the inputs. Because `coordinate_pair` and `box_size` are reversed from each other.

Reply:
To add to Dan -- we should follow standard convention, same as numpy (row x col / H x W)

Reply:
Went and corrected all of the instances of this.